### PR TITLE
Add OS column to cluster manager and node tables

### PR DESCRIPTION
**Pull request by rak-phillip**
This adds an OS column to cluster manager and node list tables by getting the operating system from node's `status.nodeInfo.operatingSystem` property.

The solution for `list/node.vue` is quite straight-forward, as we already have all the data we need after fetching our collection of nodes. I'm a little concerned that the solution for `detail/provisioning.cattle.io.cluster.vue` might not be optimal for two reasons:

1. I'm dispatching a new event, `'cluster/findAll', { type: NODE }`, that might not fit with the current "provisioning" context
2. I'm modifying our collection of machines to expose the node info we need in order to display OS.. I don't feel this is the cleanest way to get at the data. I was hoping to find a more direct way to access OS through any given machine that we might have, but I wasn't finding it

#3222

### CODE REVIEW

File: detail/provisioning.cattle.io.cluster.vue [hunk 1]
```diff
@@ -152,13 +156,24 @@ export default {
...
+          return x.spec?.clusterName === this.value.metadata.name;
+        })
+        .reduce((prev, curr) => {
+          const kubeNode = this.kubeNodes?.find(x => x.id === curr?.status?.nodeRef?.name);
+
+          curr.status.nodeInfo = kubeNode?.status?.nodeInfo;
+
+          return [
+            curr,
+            ...prev
```
Comment (richard-cox) [inline detail/provisioning.cattle.io.cluster.vue:174]: Spreading these objects will disconnect them from the base objects which gets automatically updated via socket, so should generally be avoided. As per PR comment this can be removed

File: detail/provisioning.cattle.io.cluster.vue [hunk 2]
```diff
@@ -73,12 +73,15 @@ export default {
       hash.normanNodePools = this.$store.dispatch('rancher/findAll', { type: NORMAN.NODE_POOL });
     }
 
+    hash.kubeNodes = this.$store.dispatch('cluster/findAll', { type: NODE });
```
Comment (richard-cox) [inline detail/provisioning.cattle.io.cluster.vue:76]: As per PR comment, this can be called only for RKE2 clusters

**Review by richard-cox — CHANGES_REQUESTED** [review]
There's three node(ish) lists, one is missing the OS 
- Kube Nodes `cluster/nodes` (Cluster Explorer --> Cluster --> Node)
  - OS via `status.nodeInfo.operatingSystem`
- Management Nodes `managemnet.cattle.io.node` (Cluster Management --> non-RKE2 Cluster --> Machine Pools tab)
  - _no OS column in PR_
  - OS via `status.internalNodeStatus.nodeInfo.operatingSystem`
- Machine 'nodes' `cluster.x-k8s.io.machine` (Cluster Management --> RKE2 Cluster --> Machine Pools tab)
  - Doesn't look like OS is available on root object so would need to get via the associated kube node
  - This should be done via a kubeNode property on the  `cluster.x-k8s.io.machine` model. This should be a synchronous getter (see how management.cattle.io.node model `pool` property). As this is only needed for the RKE2 case... the upfront fetch of all kube nodes in `fetch` can go in same 'else' block line 58ish

So in summary, along with inline comments,...
- add OS column to mgmt nodes list in cluster management cluster page
- use separate table OS column definitions per node type 
- for machine nodes access OS via kube node object 
- gate fetching all kube nodes given if it's a RKE2 cluster

File: list/node.vue [hunk 3]
```diff
@@ -77,13 +77,20 @@ export default {
...
-        ...RAM,
-        breakpoint: COLUMN_BREAKPOINTS.LAPTOP
-      }];
+      const headers = [
+        STATE,
+        NAME,
+        ROLES,
+        VERSION,
+        INTERNAL_EXTERNAL_IP,
+        OS,
```
[the remark this answers is not in the document]
Comment (rak-phillip) [reply list/node.vue]: I added the breakpoint only to the "Machine Nodes" because it was the most dense table. I can add to others if needed

File: detail/provisioning.cattle.io.cluster.vue (hunk 2 above)
Comment (rak-phillip) [reply detail/provisioning.cattle.io.cluster.vue]: Fixed by only fetching `kubeNodes` when we fetch the collection of machines

File: detail/provisioning.cattle.io.cluster.vue (hunk 1 above)
Comment (rak-phillip) [reply detail/provisioning.cattle.io.cluster.vue]: Reverted after adding a `kubeNode` property to `cluster.x-k8s.io.machine`

**Review by mantis-toboggan-md — APPROVED** [review]

**Comment by mantis-toboggan-md** [comment]
LGTM - verified the presence of OS column for RKE2 and non-RKE2 clusters in both cluster management and the kube node list

File: detail/provisioning.cattle.io.cluster.vue [hunk 4]
```diff
@@ -61,6 +61,7 @@ export default {
 
       if ( this.$store.getters['management/canList'](CAPI.MACHINE) ) {
         hash.machines = this.$store.dispatch('management/findAll', { type: CAPI.MACHINE });
+        hash.kubeNodes = this.$store.dispatch('cluster/findAll', { type: NODE });
```
Comment (richard-cox) [inline detail/provisioning.cattle.io.cluster.vue:64]: Ok, I missed this before, caught it now. This won't work as resources from the `cluster` store work in the context of the cluster used in the route (ish). Also applies to `kubeNode` in the model for `cluster.x-l8s.io.machine`.

So if you nav here from the cluster explorer's page for this cluster it'll be fine... but from any other location (say local cluster) this will fail.

I don't think there's a straight forward way to work around this, or fetch the OS for a `cluster.x-l8s.io.machine` without the kube node. Closest I got was `rke-machine.cattle.io.<provider>machinetemplate` which may contain an image, at least for DO it did.

@vincent99 Any suggestions?

**Review by richard-cox — CHANGES_REQUESTED** [review]

File: detail/provisioning.cattle.io.cluster.vue (hunk 4 above)
Comment (rak-phillip) [reply detail/provisioning.cattle.io.cluster.vue:64]: I can see where this solution breaks by navigating Explore Cluster => Local => More Resources => Cluster Provisioning => Clusters. I'm also seeing another unrelated error when working with an RKE cluster... should this area behave the same way as Cluster Management?

Comment (rak-phillip) [reply detail/provisioning.cattle.io.cluster.vue]: Reverted these changes until we can figure out if we can get what we need from the backend

**Review by richard-cox — APPROVED** [review]
